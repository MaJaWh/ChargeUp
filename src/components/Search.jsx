import React, { useState } from "react";
import PropTypes from "prop-types";
import "./Map";
import '../styles/search.css'
// import GetConnectorStatus from "./requests/GetConnectorStatus";

const Search = ({ setchargeSites, setChargerStatus, chargeSites }) => {
  const [searchValue, setSearchValue] = useState("");
  const [distanceValue, setDistanceValue] = useState("");
  const [ratedOutput, setRatedOutput] = useState("");

  const handleSubmit = async (event) => {
    event.preventDefault();
    getData();
  };

  const getData = async () => {
    // console.log(searchValue, distanceValue, ratedOutput);
    fetch(
      `/api/retrieve/registry/postcode/${searchValue}/dist/${distanceValue}/rated-output-kw/${ratedOutput}/format/json`
    )
      .then((res) => res.json())
      .then((data) => {
        setchargeSites(data.ChargeDevice);
        setChargerStatus(data.ChargeDevice)
        console.log(data);
      })
      .catch((err) => {
        console.log(err);
      });
  };
  return (
    <div className="search">
      <div className="search__title">Search</div>
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          placeholder="postcode"
          onChange={(e) => setSearchValue(e.target.value)}
        />
        <input
          className="search-input"
          type="text"
          placeholder="Distance from location"
          onChange={(e) => setDistanceValue(e.target.value)}
        />
      
        <input
          className="search-input"
          type="text"
          placeholder="kw/h of car"
          onChange={(e) => setRatedOutput(e.target.value)}
        />
        <button className="search-btn" type="submit">
          Find!
        </button>
        {/* <GetConnectorStatus chargeSites={chargeSites}/> */}
      </form>
    </div>
  );
};

export default Search;

Search.propTypes = {
  setchargeSites: PropTypes.func.isRequired,
};
