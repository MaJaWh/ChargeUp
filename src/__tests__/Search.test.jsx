/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";


describe("Search-snapshot", () => {
  const { asFragment } = render(<Search />);
  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});

it('contains the expected input fields', () => {
  const { getByPlaceholderText } = render(<Search />);
  expect(getByPlaceholderText('postcode')).toBeInTheDocument();
  expect(getByPlaceholderText('Distance from location')).toBeInTheDocument();
  expect(getByPlaceholderText('kw/h of car')).toBeInTheDocument();
});