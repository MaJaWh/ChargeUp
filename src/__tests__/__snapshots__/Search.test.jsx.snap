// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search-snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <div
      class="search__title"
    >
      Search
    </div>
    <form
      class="search-form"
    >
      <input
        class="search-input"
        placeholder="postcode"
        type="text"
      />
      <input
        class="search-input"
        placeholder="Distance from location"
        type="text"
      />
      <input
        class="search-input"
        placeholder="kw/h of car"
        type="text"
      />
      <div
        class="search__filters"
      >
        <div
          class="plugtype__outer-div"
        />
      </div>
      <button
        class="search-btn"
        type="submit"
      >
        Find!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
